odoo/fields.py

```diff
@@ -2068,6 +2068,28 @@ def _setup_attrs(self, model, name):
         if not self.delegate:
             self.delegate = name in model._inherits.values()
 
+    def _setup_regular_base(self, model):
+        super()._setup_regular_base(model)
+        # There are 5 possible cases here:
+        #   1. The m2o field is not required and its ondelete attribute has not been explicitly
+        #      specified, in which case it defaults to `set null`.
+        #   2. The m2o field is not required and its ondelete attribute has been explicitly
+        #      specified, in which case it is left as-is.
+        #   3. The m2o field is required and its ondelete attribute has not been explicitly
+        #      specified, in which case it defaults to `restrict`.
+        #   4. The m2o field is required and its ondelete attribute has been explicitly specified
+        #      as either `restrict` or `cascade`, in which case it is left as-is.
+        #   5. The m2o field is required and its ondelete attribute has been explicitly specified
+        #      as `set null`, this raises a ValueError as it's considered a programmer error.
```

Comment:
I would mention three cases:
1. The ondelete attribute is not set; provide a consistent default value.
2. The ondelete attribute is set to 'set null' and the field is required; this case is considered a programming error.
3. The ondelete attribute is set and consistent; nothing is done.